components/calendar/src/ethiopic.rs

```diff
@@ -0,0 +1,250 @@
+            next_year: next_year.into(),
+        }
+    }
+
+    fn debug_name() -> &'static str {
+        "Ethiopic"
+    }
+}
+
+impl Ethiopic {
+    /// Construct a new Ethiopic Calendar
+    pub fn new() -> Self {
+        Self
+    }
+
+    // "Fixed" is a day count representation of calendars staring from Jan 1st of year 1 of the Georgian Calendar.
+    // The fixed date algorithms are from
+    // Dershowitz, Nachum, and Edward M. Reingold. _Calendrical calculations_. Cambridge University Press, 2008.
+    //
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L2017
+    fn fixed_from_ethiopic(date: ArithmeticDate<Ethiopic>) -> i32 {
+        let coptic_epoch = Julian::fixed_from_julian_integers(284, 8, 29);
+        let ethiopic_epoch = Julian::fixed_from_julian_integers(8, 8, 29);
+        ethiopic_epoch - coptic_epoch
+            + Coptic::fixed_from_coptic_integers(date.year, date.month as i32, date.day as i32)
+    }
+
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L2028
+    fn ethiopic_from_fixed(date: i32) -> EthiopicDateInner {
+        let coptic_epoch = Julian::fixed_from_julian_integers(284, 8, 29);
+        let ethiopic_epoch = Julian::fixed_from_julian_integers(8, 8, 29);
+        let coptic_date = Coptic::coptic_from_fixed(date + coptic_epoch - ethiopic_epoch);
+
+        #[allow(clippy::unwrap_used)] // TODO(#1668) Clippy exceptions need docs or fixing.
+        *Date::new_ethiopic_date_from_integers(
+            coptic_date.0.year,
+            coptic_date.0.month,
+            coptic_date.0.day,
+            false,
+        )
+        .unwrap()
+        .inner()
+    }
+
+    fn days_in_year_direct(year: i32) -> u32 {
+        if Ethiopic::is_leap_year(year) {
+            366
+        } else {
+            365
+        }
+    }
+}
+
+impl Date<Ethiopic> {
+    /// Construct new Ethiopic Date
+    pub fn new_ethiopic_date_from_integers(
+        year: i32,
+        month: u8,
+        day: u8,
+        amete_alem: bool,
+    ) -> Result<Date<Ethiopic>, DateTimeError> {
+        let inner = ArithmeticDate {
+            year,
+            month,
+            day,
+            marker: PhantomData,
+        };
+
+        let bound = inner.days_in_month();
+        if day > bound {
+            return Err(DateTimeError::OutOfRange);
+        }
+
+        Ok(Date::from_raw(
+            EthiopicDateInner(inner, amete_alem),
+            Ethiopic,
+        ))
+    }
+}
+
+impl DateTime<Ethiopic> {
+    /// Construct a new Ethiopic datetime from integers
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_ethiopic_datetime_from_integers(
+        year: i32,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
+        fraction: u32,
+        amete_alem: bool,
```

Comment:
i wouldn't include this, i would default this to false always

Reply:
(also on date)

Reply:
But we plan to support both ethiopic and ethiopiaa right? Or is the idea to more clearly define and use this boolean when working with ethiopiaa?

Reply:
Yes, but most people will construct with ethiopic, and they can explicitly set amete alem if they desire.

We should add `.set_amete_alem()` to `Date<Ethiopic>` and DateTime

Reply:
`Date<Ethiopic>.inner()` doesnt seem to be mutable, so created new methods that return `Result<>`

Does this work?

Reply:
Probably can return the Date<> and DateTime<> directly instead of using a Result<>